core/textile/client.go

```diff
@@ -142,21 +156,78 @@ func (tc *textileClient) start(ctx context.Context, cfg config.Config) error {
 	tc.bucketsClient = buckets
 	tc.threads = threads
 	tc.netc = netc
+	tc.uc = getUserClient()
+	tc.ht = getHubThreadsClient()
 
 	tc.isRunning = true
 
 	// Attempt to connect to the Hub
-	_, err := tc.getHubCtx(ctx)
+	hubctx, err := tc.getHubCtx(ctx)
 	if err != nil {
 		log.Error("Could not connect to Textile Hub. Starting in offline mode.", err)
 	} else {
 		tc.isConnectedToHub = true
+
+		// setup mailbox
+		mid, err := tc.uc.SetupMailbox(hubctx)
+		if err != nil {
+			log.Error("Unable to setup mailbox", err)
+			return err
+		}
+		log.Info("Mailbox id: " + mid.String())
 	}
 
 	tc.Ready <- true
 	return nil
 }
 
+// adding for testability
+func (tc *textileClient) SetUc(uc UsersClient) {
+	tc.uc = uc
+}
+
+func getUserClient() UsersClient {
+	hubTarget := os.Getenv("TXL_HUB_HOST")
```

Comment:
@jsonsivar 

Please ignore me if I'm wrong: `TXL_HUB_HOST` or `TXL_HUB_TARGET`? It seems that we remove `TXL_HUB_HOST` from the `set-envs`.

Reply:
Good catch. Actually they should come from the global configuration object which is something I forgot to do.

Reply:
Fixed in [4d291fd](https://github.com/FleekHQ/space-daemon/pull/108/commits/4d291fdf871f31fde97bd7be5f190b43e72934a1).